database/rrdcalc.c

```diff
@@ -35,6 +35,66 @@ inline const char *rrdcalc_status2string(RRDCALC_STATUS status) {
     }
 }
 
+char *rrdcalc_replace_variables(const char *line, RRDCALC *rc)
+{
+    if (!line)
+        return NULL;
+
+    size_t pos = 0;
+    char *temp = strdupz(line);
+    char var[MAX_RRDCALC_VARIABLE];
+    int i=0;
+    char *m, *e, *lbl_value = NULL;
+
+    while ((m = strchr(temp + pos, '$'))) {
+        i=0;
+        e = m;
+        while (*e) {
+            if (*e == ' ' || i == MAX_RRDCALC_VARIABLE - 1) {
+                break;
+            }
+            else
+                var[i]=*e;
+            e++;
+            i++;
+        }
+        var[i]='\0';
+        pos = e - temp;
+        if (!strcmp(var, "$family")) {
+            char *buf = find_and_replace(temp, var, (rc->rrdset && rc->rrdset->family) ? rc->rrdset->family : "", m);
+            freez(temp);
+            temp = strdupz(buf);
+            freez(buf);
+        } else if (!strncmp(var, "$lbl_", 5)) {
```

Comment:
we may think about changing `$lbl_` to something else - I think we will have some internal labels starting with `__` - `$lbl___name` look error-prone.

Perhaps make it like a function - `$lbl_get:name`.

Reply:
This would extend a little bit the complexity of our alerts, what is fine, but it also needs a new section in our documentation.

Reply:
@ilyam8 Yes, good call, we should change it. @thiagoftsm yes, we will document it of course!